lib/logstash/outputs/kafka.rb

```diff
@@ -173,11 +177,20 @@ def register
     @producer = create_producer
     @codec.on_event do |event, data|
       begin
-        if @message_key.nil?
-          record = org.apache.kafka.clients.producer.ProducerRecord.new(event.sprintf(@topic_id), data)
-        else
-          record = org.apache.kafka.clients.producer.ProducerRecord.new(event.sprintf(@topic_id), event.sprintf(@message_key), data)
+        partition = nil
+        key = event.sprintf(@message_key) unless @message_key.nil?
+
+        timestamp = nil
+        unless @message_timestamp.nil?
+          if event.include? @message_timestamp
+            ts = event.get(@message_timestamp)
+            timestamp = ts.is_a?(LogStash::Timestamp) ? ts.time.to_i : ts.to_i
+          else
+            timestamp = event.sprintf(@message_timestamp).to_i
```

Comment:
what is the goal here? I can see this very easily causing mistakes for users. What would an expected sprintf be, here?

* `"%{+%s}%{+SSS}"` (same result as if you call `time.to_f * 1000` basically)
* Something else?

Reply:
The idea of using `sprintf` here was described in my original issue (#124), but quick, it was helpful in cases where a field containing a value in seconds could also be used as the message timestamp.

Reply:
I must have missed this comment; sorry for the delays.

If the use case is taking a numeric value as the kafka message timestamp, then I would prefer this be `event.get(@message_timestamp)` and not `event.sprintf`.